### PR TITLE
doc: Create nmstatectl man page

### PR DESCRIPTION
**Pull request by ffmancera**
Add the initial manual page for nmstatectl.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: doc/nmstatectl.8 [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+.PP
+.B show
+.RS
+Query the current network state. \fIYAML\fR is the default output format. Use
+the \fB--json\fR argument to change the output format to \fIJSON\fR. To limit
+the output state to include certain interfaces only, please specify the
+interface name. Please be advised, global config like DNS will be included.
+.PP
+For multiple interface names, use comma to separate them. You can also use
+regular expression patterns for interface names:
```
Comment (tyll) [inline doc/nmstatectl.8]: The pattern syntax does not conform to regular expressions. I would just remove the words `regular expression`. Regular expressions would be the following:

`.*`
`.`
`[seq]`
`[^seq]`

**Comment by tyll** [comment]
IMHO since we have two files with similar content, they should be generated from each other or use a common source. We could write the cli_guide in asciidoc (AFAIK it is also supported by GitHub) and generate the manpage from it: https://github.com/asciidoctor/asciidoctor.org/blob/master/docs/_includes/manpage.adoc

**Comment by ffmancera** [comment]
I am going to write the cli_guide to asciidoc and generate the manpage from it. After that, I am going to apply the changes requested.

Thanks!

**Comment by ffmancera** [comment]
Looking at other python projects manual page, they are not using the date and version field as the best way to do that is adding a Makefile for the documentation. What do you think? @cathay4t have you any suggestion on how to accomplish that?

File: doc/nmstatectl.8.in [hunk 2]
```diff
@@ -0,0 +1,91 @@
+.\" Manpage for nmstatectl.
+.TH nmstatectl 8 "@DATE@" "@VERSION@" "nmstatectl man page"
+.SH NAME
+nmstatectl \- A nmstate command line tool
+.SH SYNOPSIS
+.B nmstatectl show \fR[<\fIinterface_name\fR>] [\fB--json\fR]
```
Comment (cathay4t) [inline doc/nmstatectl.8.in]: Normally people use `[INTERFACE_NAME]`. You may check `man ls` as example.

File: packaging/nmstate.py3.spec [hunk 3]
```diff
@@ -48,6 +48,7 @@ sed -i -e '/^dbus-python$/d' requirements.txt
 %files
 %doc README.md
 %doc examples/
+%{_mandir}/man8/nmstatectl.8.gz
```
Comment (cathay4t) [inline packaging/nmstate.py3.spec]: Please use ``%{_mandir}/man8/nmstatectl.8*`, rpmbuild might compress it to other format in the future.

File: doc/nmstatectl.8.in [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+For multiple interface names, use comma to separate them. You can also use
+patterns for interface names:
+.RS
+.B *\fR matches everything
+.br
+.B ?\fR matches any single character
+.br
+.B [seq]\fR matches any character in seq
+.br
+.B [!seq]\fR matches any character not in seq
```
Comment (cathay4t) [inline doc/nmstatectl.8.in]: Please provide a example as `cli_guide` did:

For example, to show all interfaces starts with eth:
```
nmstatectl show eth\*
# The backslash is required to stop shell expanding '*' to file names.
```

File: doc/nmstatectl.8.in [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+create a checkpoint which later could be used for rollback or commit. The
+checkpoint will be the last line of \fBnmstatectl\fR output, example:
+\fI/org/freedesktop/NetworkManager/Checkpoint/1\fR.
+.IP \fB--timeout\fR=<\fItimeout\fR>
+the user must commit the changes within \fItimeout\fR, or they will be
+automatically rolled back. Default: 60 seconds.
+.SH BUG REPORTS
+Report bugs on nmstate GitHub issues <https://github.com/nmstate/nmstate>.
+.SH COPYRIGHT
+License LGPL-2.1 <https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt>.
```
Comment (cathay4t) [inline doc/nmstatectl.8.in]: The license is LGPL-2.1 or any later version.

File: doc/nmstatectl.8.in [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+By default, if the network state after state applied is not identical to the
+desired state, \fBnmstatectl\fR rollbacks to the state before \fBedit\fR
+command. Use the \fB--no-verify\fR argument to skip the verification.
+.RE
+.PP
+.B nmstatectl\fR supports manual transaction control which allows user to
+decide whether rollback to previous (before \fB"nmstatectl set/edit"\fR) state.
+.IP \fBrollback
+rollback the network state from specified checkpoint file. \fBnmstatectl\fR
+will take the last checkpoint if not defined as argument.
```
Comment (cathay4t) [inline doc/nmstatectl.8.in]: Not `last checkpoint`, `latest checkpoint`.

**Comment by ffmancera** [comment]
All changes requested applied, thanks for reviewing!

File: doc/nmstatectl.8.in [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+.RS
+change the output format to \fIJSON\fR.
+.RE
+.IP \fB--no-verify
+skip the desired network state verification.
+.IP \fB--no-commit
+create a checkpoint which later could be used for rollback or commit. The
+checkpoint will be the last line of \fBnmstatectl\fR output, example:
+\fI/org/freedesktop/NetworkManager/Checkpoint/1\fR.
+.IP \fB--timeout\fR=<\fItimeout\fR>
```
Comment (cathay4t) [inline doc/nmstatectl.8.in]: `<TIMEOUT>`

File: setup.py [hunk 8]
```diff
@@ -44,6 +45,17 @@ def get_version():
     return version
 
 
+def gen_manpage():
+    manpage = ""
+    with open('doc/nmstatectl.8.in') as f:
+        manpage = f.read()
+    manpage = manpage.replace("@DATE@", date.today().strftime("%B %d, %Y"))
+    manpage = manpage.replace("@VERSION@", get_version())
+    with open('doc/nmstatectl.8', 'w+') as f:
```
Comment (cathay4t) [inline setup.py]: `w+` is incorrect, should be `w`.

File: doc/nmstatectl.8.in [hunk 9]
```diff
@@ -0,0 +1,99 @@
+.\" Manpage for nmstatectl.
+.TH nmstatectl 8 "@DATE@" "@VERSION@" "nmstatectl man page"
+.SH NAME
+nmstatectl \- A nmstate command line tool
+.SH SYNOPSIS
+.B nmstatectl show \fR[\fIINTERFACE_NAME\fR] [\fB--json\fR]
```
Comment (cathay4t) [inline doc/nmstatectl.8.in]: the `nmstatectl show` should also be rounded with `[\fI` and `\fR`.

And the leading `\fR` seems not required.

The same also apply to the following lines.

File: doc/nmstatectl.8.in [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+command. Use the \fB--no-verify\fR argument to skip the verification.
+.RE
+.PP
+.B nmstatectl\fR supports manual transaction control which allows user to
+decide whether rollback to previous (before \fB"nmstatectl set/edit"\fR) state.
+.IP \fBrollback
+rollback the network state from specified checkpoint file. \fBnmstatectl\fR
+will take the latest checkpoint if not defined as argument.
+.PP
+.B commit
```
Comment (cathay4t) [inline doc/nmstatectl.8.in]: Should be rounded by `\fI` and \fR`.

File: doc/nmstatectl.8.in (hunk 9 above)
Comment (ffmancera) [reply doc/nmstatectl.8.in]: Looking at other manpages, they all follow that pattern. Shouldn't we do it the same way?

Examples:
nmcli(1), NetworkManager(8), iptables(8)

**Pull request merged**